File: project_euler/02/020.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const n = 100

func multiply(a []int, b int) []int {
	multi := []int{}
	carry := 0
	for i := 0; i < len(a); i++ {
		num := a[i]*b + carry
		multi = append(multi, num%10)
		carry = num / 10
	}

	for carry != 0 {
		multi = append(multi, carry%10)
		carry /= 10
	}

	return multi
}

func main() {
	fac := []int{1}
	for i := 1; i <= n; i++ {
		fac = multiply(fac, i)
	}

	sum := 0
	for i := 0; i < len(fac); i++ {
		sum += fac[i]
	}

	fmt.Printf("%d\n", sum)
}

File: project_euler/02/020.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 100


def multiply(vec_a, num_b):
    multi = []
    carry = 0
    for _, dig in enumerate(vec_a):
        tmp = dig * num_b + carry
        multi.append(tmp % 10)
        carry = tmp // 10

    while carry != 0:
        multi.append(carry % 10)
        carry //= 10

    return multi


def main():
    fac = [1]
    for i in range(1, N + 1):
        fac = multiply(fac, i)

    add_up = 0
    for _, dig in enumerate(fac):
        add_up += dig

    print(add_up)


if __name__ == '__main__':
    main()
